internal/types/path.go

```diff
@@ -69,26 +70,37 @@ type Path struct {
 	s []string
 }
 
-// NewPath returns Path from a strings slice.
-func NewPath(path ...string) Path {
+// NewStaticPath returns Path from a strings slice.
+//
+// It panics on invalid paths. For that reason, it should not be used with user-provided paths.
```

Comment:
Can we explain what paths we consider invalid?

Reply:
Sure, in #1954